specs/Extensions.md

```diff
@@ -0,0 +1,585 @@
+    /// the browser extension API [`chrome.runtime.id`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/id). 
+    /// Please see that documentation for more details on how the ID is generated.
+    /// The caller must free the returned string with `CoTaskMemFree`.  See
+    /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+    [propget] HRESULT Id([out, retval] LPWSTR* value); 
+    /// This is the browser extension's name. This value is defined in this browser extension's manifest.json file. 
+    /// If manifest.json define extension's localized name, this value will be the localized version of the name.
+    /// Please see [Manifest.json name](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/name) for more details.
+    /// The caller must free the returned string with `CoTaskMemFree`.  See
+    /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+    [propget] HRESULT Name([out, retval] LPWSTR* value);
+    /// Removes this browser extension from its WebView2 Profile. The browser extension is removed immediately including from all 
+    /// currently running HTML documents associated with this WebView2 Profile. The removal is persisted and future uses of this profile 
+    /// will not have this extension installed. After an extension is removed, calling `Remove` again will cause an exception.
+    HRESULT Remove([in] ICoreWebView2BrowserExtensionRemoveCompletedHandler* handler);
+    /// If isEnabled is true then the Extension is enabled and running in WebView instances.
+    /// If it is false then the Extension is disabled and not running in WebView instances.
+    /// When a Extension is first installed, `IsEnable` are default to be `TRUE`.
+    [propget] HRESULT IsEnabled([out, retval] BOOL* value);
+    /// Sets whether this browser extension is enabled or disabled. This change applies immediately 
+    /// to the extension in all HTML documents in all WebView2s associated with this profile.
+    /// After an extension is removed, calling `SetIsEnabled` will not change the value of `IsEnabled`.
+    HRESULT SetIsEnabled([in] BOOL isEnabled, [in] ICoreWebView2BrowserExtensionSetEnabledCompletedHandler* handler);
+}
+
+/// Provides a set of properties for managing browser Extension Lists. This
+/// includes the number of browser Extensions in the list, and the ability
+/// to get an browser Extension from the list at a particular index.
+[uuid(59251055-F2F1-448F-A096-F996FB9ACBE2), object, pointer_default(unique)]
+interface ICoreWebView2BrowserExtensionList : IUnknown {
+    /// The number of browser Extensions in the list.
+    [propget] HRESULT Count([out, retval] UINT* count);
+    /// Gets the browser Extension located in the browser Extension List at the given index.
+    HRESULT GetValueAtIndex([in] UINT index, [out, retval] ICoreWebView2BrowserExtension** extension);
```

Comment:
Checking: this can't change from another thread, so there's no chance that this index gets out of sync with the Count?